docs/rules.md

```diff
@@ -13,6 +13,7 @@
 * [no-attrs-in-components](rule/no-attrs-in-components.md)
 * [no-bare-strings](rule/no-bare-strings.md)
 * [no-debugger](rule/no-debugger.md)
+* [no-dom-events](rule/no-dom-events.md)
```

Comment:
Not sure where to do it, so I picked here 😄 

We should bikeshed the name of this rule a bit. We certainly don't want to infer that "dom events" are bad (indeed `{{action` actually _uses_ DOM events). Some names off the cuff that strike a better balance:

* `no-element-event-callbacks`
* `no-element-property-events`

Reply:
I renamed it to `no-element-event-attribute-actions`. How does that sound?

Reply:
I worry about muddying the waters between attributes and properties even further. The specific thing that we are trying to avoid is setting these as _properties_ (not attributes). This is because Ember (_currently_) uses a "properties first" mentality.

<details>
<summary>Specifically we do _something_ akin to this...</summary>

```js
let normalizedPropertiesByTag = new Map();

function selectAttributeOrPropertyFor(element, name) {
  let propertiesForTag = normalizedPropertiesByTag.get(element.tagName);
  
  if (propertiesForTag === undefined) {
    let propertiesForTag = new Map();
    for (let key in element) {
      propertiesForTag.set(key.toLowerCase(), key);
    }
    normalizedPropertiesByTag.set(element.tagName, propertiesForTag);
  }

  let propertyName = propertiesForTag.get(name.toLowerCase());
  if (propertyName) {
    return { type: 'property', name: propertyName };
  }

  return { type: 'attribute', name };
}

function populateAttributeOrProperty(element, attributeOrProperty, value) {
  let details = selectAttributeOrPropertyFor(element, attributeOrProperty);

  if (details.type === 'property') {
    element[details.name] = value;
  } else {
    element.setAttribute(details.name, value);
  }
}

// when rendering for <div data-foo="asdf"></div>
let el = document.createElement('div');
populateAttributeOrProperty(el, 'data-foo', 'asdf');
```
</details>
<br>

As you dig into the difference between attributes and properties, you see that the specific thing that we are trying to avoid is using `onclick` as a _property_ **not** an attribute. 

You can see that by testing this out:

```js
  let el = document.createElement("button");

  // setup the button text
  let text = document.createTextNode("Click Me -- Attribute!");
  el.appendChild(text);

  // trying to set a "real" function to an attribute
  el.setAttribute("onclick", () => alert("clicked -- attribute!"));
  document.body.appendChild(el);
```

Note that the alert does not fire. If you inspect the DOM you will see that what is done is that the function is `.toString()`'ed before appending to the attribute value. The only way to preserve the actual function is if we set as a property!

(I made a [codepen here](https://codepen.io/rwjblue/pen/wNKKrY) to make it easier to test this out).

Reply:
How do you feel about `no-element-event-actions` or `no-element-event-property-actions`?

Reply:
Thanks for the explanation of the difference! I'm fine with either name you suggested. I renamed it to `no-element-event-actions`. I also updated all usages of `attribute` to `property` throughout the files for the new rule.